src/providers/hana/qgshanaprovider.cpp

```diff
@@ -1336,19 +1337,36 @@ QgsRectangle QgsHanaProvider::estimateExtent() const
   if ( conn.isNull() )
     return QgsRectangle();
 
+  if ( QgsHanaUtils::toHANAVersion(conn->getDatabaseVersion()).majorVersion() < 2 )
+    useEstimatedMetadata = false;
+
   try
   {
     QString sql;
-    if ( isSrsRoundEarth( *conn, mSrid ) )
+
+    if ( useEstimatedMetadata )
```

Comment:
The try/catch block became even larger. It makes sense to move all code related to the sql string out of this block.

Reply:
Done.